### PR TITLE
[MICA-15859] fix exposure to xml-crypto vulnerability

### PR DESCRIPTION
**Pull request by alwhiting**
Please see comment below.

### CODE REVIEW

**Review by colefranz — APPROVED** [review]
LGTM based on what I know from here, but I don't much understand the test change https://github.com/node-saml/xml-crypto/discussions/399

hoping to leave the REAL approving to @nsvarich

File: test/saml2.coffee [hunk 1]
```diff
@@ -223,7 +223,7 @@ describe 'saml2', ->
         assert.notEqual null, result
 
       it 'accepts signed AD FS 2019 xml', ->
-        result = saml2.check_saml_signature(get_test_file("adfs-assertion-response.xml"), get_test_file("adfs-b64-key.txt"))
+        result = saml2.check_saml_signature(get_test_file("adfs-assertion-response.xml"), get_test_file("adfs-cert.b64.txt"))
         assert.deepEqual result, [get_test_file("adfs-assertion-response-signed-data.xml")]
 
       it 'rejects AD FS 2019 xml with invalid signature', ->
```
Comment (nsvarich) [inline test/saml2.coffee:229]: Just for clarity could you update this related invalid test too to use the cert `get_test_file("adfs-cert.b64.txt")` just so it's testing with the proper 2nd arg?

Comment (alwhiting) [reply test/saml2.coffee:229]: 👍

**Review by nsvarich — APPROVED** [review]
LGTM, from this in the [xml-crypto README](https://github.com/node-saml/xml-crypto) confirming your fix:

> The certificate that will be used to check the signature will first be determined by calling this.getCertFromKeyInfo(), which function you can customize as you see fit. If that returns null, then publicCert is used

**Pull request merged**